assets/js/util/index.js

```diff
@@ -185,20 +172,68 @@ export const readableLargeNumber = ( number, currencyCode = false ) => {
 };
 
 /**
- * Formats a number using the Internationalization Number Format API.
+ * Formats a number with unit using the JS Internationalization Number Format API.
  *
- * @since 1.0.0
- * @see {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/NumberFormat/NumberFormat|`options` parameter} For all available formatting options.
+ * @since n.e.x.t
  *
- * @param {number} number           The number to format.
- * @param {Object} [options]        Formatting options.
- * @param {string} [options.locale] Locale to use for formatting. Defaults to current locale used by Site Kit.
- * @return {string} The formatted number.
+ * @param {number|string}            number    The number to format.
+ * @param {string}                   [unit]    The unit for the number.
+ * @param {Intl.NumberFormatOptions} [options] Formatting options.
+ * @return {string} The formatted number with unit.
  */
-export const numberFormat = ( number, options = {} ) => {
-	const { locale = getLocale(), ...formatOptions } = options;
+export const numFmt = ( number, unit, options = {} ) => {
```

Comment:
@kostyalmm See the IB, the signature should be `numFmt( number, options = {} )` in order to match the familiar `numberFormat` function. The `unit` you have here should instead simply be a shorthand version of `options`, to be able to call the function in simpler ways for the most common cases.

Please change this to the signature with the two parameters similar to the preliminary code in the IB.

Reply:
will we be overloading the `options` parameter as `{object|string}`? That seemed a bit odd to me.
What will happen when we want to use percentage and also pass additional option such as `maximumFractionDigits`?
I made this change keeping this in mind. Please let me know if I should revert this to the IB.

Reply:
Yes, the `string` is just a shorthand, but yes it should be overloaded in that sense. Since it's just a shorthand, if you wanted to provide more details, you could still pass an object instead with `style: 'percent'` and other options in that case.